packages/flutter_tools/test/general.shard/windows/visual_studio_validator_test.dart

```diff
@@ -19,19 +19,19 @@ void main() {
 
     setUp(() {
       mockVisualStudio = MockVisualStudio();
-      // Mock a valid VS installation.
-      when(mockVisualStudio.isInstalled).thenReturn(true);
+      // Default values for a missing installation.
```

Comment:
Let's make this more explicit, to help avoid this category of bug in the future:
- Make a configureMockVisualStudioAsNotInstalled() and move all this there
- Make a configureMockVisualStudioAsInstalled() which sets up basic settings for installed (it can call the above as a starting point if that's easier, as an implementation detail)
- Have every test start by calling one of the two.

Reply:
Yeah I started like that but then thought it might be easier with start with one default, but I agree there is less room for error this way